src/FFmpegReader.cpp

```diff
@@ -706,9 +706,12 @@ void FFmpegReader::UpdateVideoInfo() {
 	info.vcodec = pCodecCtx->codec->name;
 	info.video_bit_rate = (pFormatCtx->bit_rate / 8);
 
-	// set frames per second (fps)
-	info.fps.num = pStream->avg_frame_rate.num;
-	info.fps.den = pStream->avg_frame_rate.den;
+	// Frame rate from the container and codec
+	AVRational framerate = av_guess_frame_rate(pFormatCtx, pStream , NULL);
+	info.fps.num = framerate.num;
+	info.fps.den = framerate.den;
+
+	ZmqLogger::Instance()->AppendDebugMethod("FFmpegReader::UpdateVideoInfo", "fps_num", info.fps.num, "fps_den", info.fps.den, "avg_fps_num", pStream->avg_frame_rate.num, "avg_fps_den", pStream->avg_frame_rate.den);
```

Comment:
For testing/comparison purposes during development I totally see the value, but in terms of the library's logging going forward, it doesn't really make sense to log values we're _not_ using (`pStream->avg_frame_rate_{num,den}`). If we're switching to `av_guess_frame_rate()` because it produces better values (and if you say that it does, good enough for me), then IMHO we should just log **those** values. Also, tweaked the labels of logged values to reflect how they're stored in the code.
```suggestion
	ZmqLogger::Instance()->AppendDebugMethod("FFmpegReader::UpdateVideoInfo (set info.fps)", "info.fps.num", info.fps.num, "info.fps.den", info.fps.den);
```

Reply:
(If we want to see comparisons in the short-term, to "monitor" the accuracy of `av_guess_frame_rate()`, then I'd say let's log the `pstream->avg_frame_rate` values in a **separate** log message. Later, we can just delete that line.)
```cpp
ZmqLogger::Instance()->AppendDebugMethod("FFmpegReader::UpdateVideoInfo (pStream->avg_frame_rate)", "num", pStream->avg_frame_rate.num, "den", pStream->avg_frame_rate.den);
```

Reply:
Of course, message can be omitted completely. This strings is called rarely (on file open only, thus when seeking will be fixed - this will be called just once per file) and I placed debugging info just for comparison. The clip FPS has huge impact on everything else in OpenShot (start/duration etc.) that calculated in seconds, so this change requires a bit more attention. **I think**, this solution is better than the one the OpenShot has right now.

I'm choosing the message split.